fe/src/main/java/org/apache/doris/external/EsStateStore.java

```diff
@@ -17,94 +17,95 @@
 import org.json.JSONObject;
 
-import java.io.IOException;
-import java.util.Collections;
-import java.util.Comparator;
-import java.util.List;
-import java.util.Map;
-import java.util.stream.Collectors;
-
-import okhttp3.Authenticator;
-import okhttp3.Call;
-import okhttp3.Credentials;
-import okhttp3.OkHttpClient;
-import okhttp3.Request;
-import okhttp3.Response;
-import okhttp3.Route;
-
 
 /**
  * it is used to call es api to get shard allocation state
  */
 public class EsStateStore extends MasterDaemon {
+
     private static final Logger LOG = LogManager.getLogger(EsStateStore.class);
 
     private Map<Long, EsTable> esTables;
 
+    private Map<Long, EsRestClient> esClients;
+
     public EsStateStore() {
         super("es state store", Config.es_state_sync_interval_second * 1000);
         esTables = Maps.newConcurrentMap();
+        esClients = Maps.newConcurrentMap();
     }
 
     public void registerTable(EsTable esTable) {
         if (Catalog.isCheckpointThread()) {
             return;
         }
         esTables.put(esTable.getId(), esTable);
-        LOG.info("register a new table [{}] to sync list", esTable.toString());
+        esClients.put(esTable.getId(),
+            new EsRestClient(esTable.getSeeds(), esTable.getUserName(), esTable.getPasswd()));
+        LOG.info("register a new table [{}] to sync list", esTable);
     }
 
     public void deRegisterTable(long tableId) {
         esTables.remove(tableId);
+        esClients.remove(tableId);
         LOG.info("deregister table [{}] from sync list", tableId);
     }
-    
+
     @Override
     protected void runAfterCatalogReady() {
         for (EsTable esTable : esTables.values()) {
             try {
-                EsRestClient client = new EsRestClient(esTable.getSeeds(),
-                        esTable.getUserName(), esTable.getPasswd());
-                // if user not specify the es version, try to get the remote cluster versoin
-                // in the future, we maybe need this version
-                String indexMetaData = client.getIndexMetaData(esTable.getIndexName());
-                if (indexMetaData == null) {
+                EsRestClient client = esClients.get(esTable.getId());
+
+                if (esTable.isKeywordSniffEnable() || esTable.isDocValueScanEnable()) {
+                    JSONObject properties = client.getIndexProperties(esTable.getIndexName(), esTable.getMappingType());
+                    if (properties == null) {
+                        continue;
+                    }
+                    setEsTableContext(properties, esTable);
+                }
+
+                EsIndexState esIndexState = client.getIndexState(esTable.getIndexName());
+                if (esIndexState == null) {
                     continue;
                 }
-                EsTableState esTableState = parseClusterState55(indexMetaData, esTable);
+
+                EsTableState esTableState = setTableStatePartitionInfo(esTable, esIndexState);
```

Comment:
`setTableStatePartitionInfo` is so long and maybe not reflect the meaning of this func.

Reply:
This is an old piece of code that I'm not familiar with